src/Modules/MotdModule/Services/MotdService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Timers;
+using EvoSC.Common.Interfaces.Models;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.MotdModule.Interfaces;
+using Microsoft.Extensions.Logging;
+using Timer = System.Timers.Timer;
+
+namespace EvoSC.Modules.Official.MotdModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class MotdService : IMotdService, IDisposable
+{
+    public static readonly string ErrorTextMotdNotLoaded = "Motd couldn't be fetched.";
+    
+    private readonly IManialinkManager _manialink;
+    private readonly IHttpService _httpService;
+    private readonly IMotdRepository _repository;
+    private readonly ILogger<MotdService> _logger;
+
+    private readonly Timer _motdUpdateTimer;
+    
+    private string _motdUrl;
+    private int _timerInterval;
+
+    private string MotdText { get; set; } = "";
+    
+    public bool IsDisposed { get; private set; }
+
+    public MotdService(IManialinkManager manialink, IHttpService httpService, 
+        IMotdRepository repository, IMotdSettings motdSettings, ILogger<MotdService> logger)
+    {
+        _manialink = manialink;
+        _httpService = httpService;
+        _repository = repository;
+        _motdUrl = motdSettings.MotdUrl;
+        _timerInterval = motdSettings.MotdFetchInterval;
+        _logger = logger;
+        
+        _motdUpdateTimer = new Timer
+        {
+            Interval = 1,
+            Enabled = true,
+            AutoReset = true
+        };
+        _motdUpdateTimer.Elapsed += MotdUpdateTimerOnElapsed;
+    }
+
+    private void MotdUpdateTimerOnElapsed(object? sender, ElapsedEventArgs e)
+    {
+        Timer timer = (Timer)sender!;
+
+        timer.Interval = _timerInterval;
+        MotdText = GetMotdAsync().Result;
+        _logger.LogDebug($"Fetching ");
+    }
+    
+    public void SetInterval(int interval)
+    {
+        _timerInterval = interval;
+        _motdUpdateTimer.Interval = interval;
+    }
+
+    public void SetUrl(string url)
+    {
+        _motdUrl = url;
+        MotdText = GetMotdAsync().Result;
+        if (!_motdUpdateTimer.Enabled)
+        {
+            _motdUpdateTimer.Enabled = true; // re-enable the timer when the url updates.
+        }
+    }
+
+    public async Task ShowAsync(IPlayer player)
+    {
+        var isCheckboxChecked = (await _repository.GetEntryAsync(player))?.Hidden ?? false;
+        await _manialink.SendManialinkAsync(player, "MotdModule.MotdTemplate", new { isChecked = isCheckboxChecked, text = MotdText });
+    }
+    
+    public async Task<string> GetMotdAsync()
+    {
+        try
+        {
+            return await _httpService.GetAsync(_motdUrl);
+        }
+        catch (InvalidOperationException ex)
+        {
+            _logger.LogError($"Motd couldn't be fetched from url \"{_motdUrl}\"");
```

Comment:
```suggestion
            _logger.LogError(ex, "Motd couldn't be fetched from url {MotdUrl}", _motdUrl);
```

Use structured logging instead of string interpolation: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-7.0#log-message-template

More info on what structured logging allows for and why to use it, if you're interested:
- https://github.com/NLog/NLog/wiki/How-to-use-structured-logging
- https://messagetemplates.org/

For us, this benefits our backend system when we switch the logging output to JSON, makes it easier to query logs.

Also provide the exception object, it will print out the stack trace and message automatically: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-7.0#log-exceptions